xcdat/temporal.py

```diff
@@ -611,48 +639,88 @@ def departures(
             'drop_incomplete_djf': 'False'
         }
         """
-        ds = self._dataset.copy()
+        # 1. Set the attributes for this instance of `TemporalAccessor`.
+        # ----------------------------------------------------------------------
+        self._set_arg_attrs(
+            "departures", freq, weighted, reference_period, season_config
+        )
         self._set_data_var_attrs(data_var)
-        self._set_arg_attrs("departures", freq, weighted, season_config)
 
-        # Preprocess the dataset based on method argument values.
+        # 2. Copy the original dataset and preprocess (if needed) for reuse.
+        # ----------------------------------------------------------------------
+        ds = self._dataset.copy()
+
+        # Preprocess ahead of time to avoid needing to preprocess again when
+        # calling the group average or climatology APIs in step #3.
         ds = self._preprocess_dataset(ds)
 
-        # Group the observation data variable.
-        dv_obs = _get_data_var(ds, data_var)
-        self._labeled_time = self._label_time_coords(dv_obs[self.dim])
-        dv_obs_grouped = self._group_data(dv_obs)
+        # 3. Calculate the grouped average and climatology of the data variable.
+        # ----------------------------------------------------------------------
+        # The climatology and grouped average APIs are called on the copied
+        # dataset to create separate instances of the `TemporalAccessor` class.
+        # This is done to avoid overriding the attributes of the current
+        # instance of `TemporalAccessor` (set in step #1 above).
+
+        # Group averaging is only required if the dataset's frequency (input)
+        # differs from the `freq` arg (output).
+        ds_obs = ds.copy()
+        inferred_freq = self._infer_freq()
+        if inferred_freq != freq:
+            ds_obs = ds_obs.temporal.group_average(
+                data_var,
+                freq,
+                weighted,
+                keep_weights,
+                season_config,
+            )
```

Comment:
Added conditional for group averaging only if the input and output frequencies are the same. This should save on performance costs for this case.

Related to https://github.com/xCDAT/xcdat/pull/417#discussion_r1113628929 and https://github.com/xCDAT/xcdat/pull/417#pullrequestreview-1308232306.